lib/mobility-core/src/Kernel/Prelude.hs

```diff
@@ -137,3 +137,8 @@ infixl 1 =<<<
 
 hoistMaybe :: Applicative m => Maybe b -> MaybeT m b
 hoistMaybe = MaybeT . pure
+
+lastMaybe :: [a] -> Maybe a
+lastMaybe [] = Nothing
+lastMaybe [x] = Just x
```

Comment:
you can use `lastMaybe list_name = Just (last list_name)`

Reply:
`last` is an unsafe function so avoided it

Reply:
I think it is unsafe only when the list empty, which you allready have handled

Reply:
`lastMaybe :: [a] -> Maybe a
lastMaybe = lastMay`

this is working fine.